src/cmds/pbs_rdel.c

```diff
@@ -120,14 +120,15 @@ main(int argc, char **argv, char **envp)
 			any_failed = 1;
 			continue;
 		}
-		connect = cnt2server(server_out);
 
+		connect = cnt2server(server_out);
 		if (connect <= 0) {
 			fprintf(stderr, "pbs_rdel: cannot connect to server %s (errno=%d)\n",
 				pbs_server, pbs_errno);
 			any_failed = pbs_errno;
 			continue;
-		}
+		} else
+			show_svr_inst_fail(connect, argv[0]);
```

Comment:
How about moving the call to 'show_svr_inst_fail' inside cnt2server()? That way it'll be guaranteed to be called for every client command

Reply:
We could have unless there are special checks in some clients which do not need this message to be printed. For instance, if a client is invoked in the quiet mode it is not expecting any message in stderr.

Reply:
and there could be client-specific error handling in the future than just printing a log to the console. we are not there yet.

Reply:
i didn't know we had a quiet mode :) The other thing we can do is have cnt2server return a specific PBSE_ code when one of the instances fail to connect, and then the clients can print that error in the if (connect <= 0) {} block itself. I think there's already such a code actually. What do you think?

Reply:
but connect will be >=0 if one of the instance gets connected.
Also, pbs_errno will get overridden with each new connection.

Reply:
here's what I'm suggesting:
- cnt2server_extend() calls a function to check if any of the instances failed, even though pbs_connect_extend returned > 0
- if an instance failed, cnt2server sets pbs_errno to a PBSE_ code which will tell clients that one of the instances is down. But it will still return the cluster fd, not -1
- Clients will then check if (connect < 0 || pbs_errno == PBSE_) {print connection error, if needed, exit from program}

That way cnt2server handles the check for any instance connection failures and client decides what to do with it.

Reply:
> if needed, exit from program

I'm not sure whether I understand you correctly. That again requires another condition to be added which is similar to what we have now. So what is the advantage of this approach?
We don't want to exit from the program if one of the instances is connected.

To address the readability concerns, I'm calling show_svr_inst_failure only if pbs_errno is set. See: 97cc87e3b918de5101b19e658c98834c689940ed